src/lib.rs

```diff
@@ -398,7 +409,8 @@ impl<'a, T> Cursor<'a, T> {
                 Some(next) => {
                     self.prev = Raw::some(&mut **next);
                     unsafe {
-                        Some(mem::copy_mut_lifetime(self, &mut next.elem))
+                        // upgrade the lifetime
+                        Some(mem::transmute(&mut next.elem))
```

Comment:
You can upgrade lifetimes without transmute, it's a bit wordy though! `&mut *(&mutnext.elem as *mut _)`   something like that. Either way is fine to me, since it's a bit much.